src/Services/GTT/shared/GTT.Infrastructure/Repositories/ChallengeRepository.cs

```diff
@@ -1,72 +1,58 @@
-
-            return result;
+            _connection = dbConnectionFactory.CreateConnection();
         }
-
-        public async Task<int> DeleteAsync(int id)
+        public ChallengeRepository(IDbConnection dbConnection) 
         {
-            var insertChallengeSql = @"DELETE FROM Products WHERE Id = @Id";
-            var result = await _connection.ExecuteAsync(insertChallengeSql, new { Id = id }, _tran);
-
-            return result;
+            _connection = dbConnection;
         }
 
-        public async Task<IReadOnlyList<Challenge>> GetAllAsync()
+        public async Task<ChallengeVM> AddAsync(CreateChallengeData challenge)
         {
-            var insertChallengeSql = @"SELECT * FROM Challenge";
-            var result = await _connection.QueryAsync<Challenge>(insertChallengeSql, _tran);
-
-            return result.ToList();
-        }
-
-        public async Task<Challenge> GetByIdAsync(int id)
-        {
-            var insertChallengeSql = @"SELECT * FROM Challenge WHERE Id = @Id";
-            var result = await _connection.QuerySingleOrDefaultAsync<Challenge>(insertChallengeSql, new { Id = id }, _tran);
-
-            return result;
-        }
-
-        public async Task<int> UpdateAsync(Challenge entity)
-        {
-            var sql = @"UPDATE Challenge SET something equal something";
-            var result = await _connection.ExecuteAsync(sql, entity, _tran);
-
-            return result;
+            try
+            {
+                var insertChallengeSql = @"INSERT INTO Challenge(Calories, SplatPoints, AvgHR, MaxHR, Miles, Steps, MemberID, ClassID, CreatedDate, UpdatedDate, CreatedBy, UpdatedBy)
+                                    VALUES(@Calories, @SplatPoints, @AvgHR, @MaxHR, @Miles, @Steps, @MemberID, @ClassID, @CreatedDate, @UpdatedDate, @CreatedBy, @UpdatedBy)
+                                    DECLARE @challengeID int
+                                    SET @challengeID = SCOPE_IDENTITY()
+                                    SELECT* FROM Challenge WHERE challengeID = @ChallengeId
+                                    ";
+
+                var param = new
+                {
+                    Calories = challenge.Calories,
+                    SplatPoints = challenge.SplatPoints,
+                    AvgHR = challenge.AvgHr,
+                    MaxHR = challenge.MaxHr,
+                    Miles = challenge.Miles,
+                    Steps = challenge.Steps,
+                    MemberID = challenge.memberID,
+                    ClassID = challenge.classID,
+                    CreatedDate = challenge.CreatedDate,
+                    UpdatedDate = challenge.UpdatedDate,
+                    CreatedBy = challenge.CreatedBy,
+                    UpdatedBy = challenge.UpdatedBy,
+                };
+
+                var result = await _connection.QueryFirstAsync<ChallengeVM>(insertChallengeSql, param);
```

Comment:
I think we don't need to return the object, just need return status and code

Reply:
You can follow my PR feat/88-Excercise-Group in file ExcerciseGroupRepository